### PR TITLE
feat(omron_node): diagnostics for frequency

### PR DESCRIPTION
**Pull request by reinzor**

### CODE REVIEW

File: omron_os32c_driver/src/os32c_node.cpp [hunk 1]
```diff
@@ -113,6 +123,10 @@ int main(int argc, char *argv[])
       laserscan_msg.header.seq++;
       laserscan_pub.publish(laserscan_msg);
 
+      // Tick the frequency and update the diagnostics
+      frequency_status.tick();
+      updater.update();
```
Comment (mikepurvis) [inline omron_os32c_driver/src/os32c_node.cpp]: Any reason to do this separately vs. using DiagnosedPublisher?

http://docs.ros.org/api/diagnostic_updater/html/classdiagnostic__updater_1_1DiagnosedPublisher.html

File: omron_os32c_driver/src/os32c_node.cpp [hunk 2]
```diff
@@ -41,6 +43,8 @@ using eip::socket::TCPSocket;
 using eip::socket::UDPSocket;
 using namespace omron_os32c_driver;
 
+double RATE = 12.856;
```
Comment (mikepurvis) [inline omron_os32c_driver/src/os32c_node.cpp]: Parameterize the expected rate and permitted deviation, please.

File: omron_os32c_driver/src/os32c_node.cpp (hunk 1 above)
Comment (reinzor) [reply omron_os32c_driver/src/os32c_node.cpp]: Thanks for the review, I was not aware of the `DiagnosedPublisher`, I will refactor this to the diagnosed publisher

File: omron_os32c_driver/src/os32c_node.cpp (hunk 2 above)
Comment (reinzor) [reply omron_os32c_driver/src/os32c_node.cpp]: Ok, any suggestions for naming? 

- `~expected_frequency` (default=`12.856`)
- `~frequency_tolerance` (default=`0.1`)
- `~timestamp_min_acceptable` (default=`-1`)
- `~timestamp_max_acceptable` (default=`5`)

**Comment by reinzor** [comment]
Can this one be merged?

**Comment by reinzor** [comment]
ping

**Comment by mikepurvis** [comment]
Looks great, thanks!

Note that I'm not actively using or maintaining this driver. If you're using it, are you interested in becoming the maintainer?

**Comment by reinzor** [comment]
Thanks Mike, I would be interested. Could you give me some pointers what it means to "maintain" a package. Does it also include https://github.com/ros-drivers/odva_ethernetip ?

**Comment by mikepurvis** [comment]
Yes, you could definitely have both repos. I would give you permission to write them on github as well as their corresponding GBP repos for the purposes of creating releases.

See for more details: http://wiki.ros.org/MaintenanceGuide

**Comment by reinzor** [comment]
Thanks. Sure I will maintain the packages. Thanks for the information regarding the responsibilities.

**Pull request merged**